CLDSRV-792: Split bucket logging API control from writing server-access.logs file

dvasilas opened the pull request:
Adds independent control over bucket logging API availability, separate from server access logging using an enum.

dvasilas committed:
CLDSRV-792: Bump package.json version to 9.2.4

bert-e commented:
# Hello dvasilas,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/cloudserver/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-792 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `9.2.4`


Please check the `Fix Version/s` of CLDSRV-792, or the target
branch of this pull request.

leif-scality reviewed (changes requested):
I think we should do something like this
```
serverAccessLogs.enabledConfigRoutes: bool // Returns NotImplemented if false
serverAccessLogs.enabled: bool // Doesn't write to the file if false
```
You can disable the logging in `server.js`

BourgoisMickael reviewed (commented):
Why would we write the log file without having the put-bucket-logging enabled ?

This is kind of confusing, according to your table, with 2 booleans and 1 combination that's a misconfig, it looks like a tri-state / three-state.

Should you rather have 1 `state` or `mode` variable with an enum value:
- `DISABLED` (or absence of the config)
- `LOG` or `LOG_ONLY`
- `API` or `FULL` or `ENABLED`

dvasilas commented:
> Why would we write the log file without having the put-bucket-logging enabled ?

Because we use the same file for s3 analytics.

The default state when installing S3C should be
- We write the file so that s3 analytics can use it
- Bucket logging APIs are disabled

Having an enum makes sense indeed, rather than using 2 variables to represent 3 state.
I will go with that.

dvasilas committed:
CLDSRV-792: Refactor serverAccessLogs to use mode enum

Replace the enabled boolean with a mode enum field to better
represent the three valid states of the feature.

The mode field supports three values:
- 'disabled': No logging at all
- 'log-only': Logging to server-access file, but bucket logging APIs return NotImplemented
- 'enabled':  Logging to server-access file, and bucket logging APIs
  available

leif-scality reviewed (approved):
Would have used constants instead of strings, but if not the rest is ok

dvasilas commented:
> Would have used constants instead of strings.

Done

BourgoisMickael commented on lib/Config.js:
```diff
@@ -574,9 +574,20 @@ function parseIntegrityChecks(config) {
+const serverAccessLogsModes = {
+    DISABLED: 'disabled',
+    LOG_ONLY: 'log-only',
+    ENABLED: 'enabled',
```
```suggestion
    DISABLED: 'DISABLED',
    LOG_ONLY: 'LOG_ONLY',
    ENABLED: 'ENABLED',
```

I'd prefer we keep same value and key for enums

dvasilas replied:
Done.

BourgoisMickael commented on lib/Config.js:
```diff
@@ -574,9 +574,20 @@ function parseIntegrityChecks(config) {
+        serverAccessLogsModes.DISABLED,
+        serverAccessLogsModes.LOG_ONLY,
+        serverAccessLogsModes.ENABLED,
+    ];
```
```suggestion
    const validModes = Object.values(serverAccessLogsModes);
```

dvasilas replied:
:+1:

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

dvasilas commented:
/approve

bert-e commented:
# Build failed



The [build](https://github.com/scality/cloudserver/actions?query=branch%3Aimprovement/CLDSRV-792) for [commit](https://github.com/scality/cloudserver/commit/2c791e25f692cc4eac1fac471d020d012945195a) did not succeed in branch [improvement/CLDSRV-792](https://github.com/scality/cloudserver/compare/development/9.2...improvement/CLDSRV-792)



*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/9.2`



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.8`
* `development/9.0`
* `development/9.1`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/9.2`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.8`
* `development/9.0`
* `development/9.1`



Please check the status of the associated issue CLDSRV-792.

Goodbye dvasilas.
